tools/dynamicdatamonitor/DynamicDataMonitor.cpp

```diff
@@ -0,0 +1,235 @@
+/*
+ *
+ *
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#include <dds/DCPS/transport/framework/TransportRegistry.h>
+#include <dds/DCPS/transport/framework/TransportConfig.h>
+#include <dds/DCPS/transport/framework/TransportInst.h>
+
+#include <dds/DCPS/Recorder.h>
+#include <dds/DdsDcpsInfrastructureC.h>
+#include <dds/DCPS/Marked_Default_Qos.h>
+#include <dds/DCPS/Service_Participant.h>
+#include <dds/DCPS/SubscriberImpl.h>
+#include <dds/DCPS/WaitSet.h>
+#include <dds/DCPS/Serializer.h>
+#include <dds/DCPS/XTypes/TypeObject.h>
+#include <dds/DCPS/XTypes/DynamicData.h>
+#include "dds/DCPS/StaticIncludes.h"
+#if defined ACE_AS_STATIC_LIBS && !defined OPENDDS_SAFETY_PROFILE
+#include <dds/DCPS/transport/udp/Udp.h>
+#include <dds/DCPS/transport/multicast/Multicast.h>
+#include <dds/DCPS/RTPS/RtpsDiscovery.h>
+#include <dds/DCPS/transport/rtps_udp/RtpsUdp.h>
+#include <dds/DCPS/transport/shmem/Shmem.h>
+#endif
+
+#include <ace/Arg_Shifter.h>
+#include <ace/Semaphore.h>
+#include <ace/Thread_Semaphore.h>
+
+#include <iostream>
+#include <sstream>
+
+long num_samples = 0;
+long num_seconds = 0;
+// parse the command line arguments
+int parse_args (int& argc, ACE_TCHAR *argv[])
+{
+  u_long mask =  ACE_LOG_MSG->priority_mask(ACE_Log_Msg::PROCESS);
+  ACE_LOG_MSG->priority_mask(mask | LM_TRACE | LM_DEBUG, ACE_Log_Msg::PROCESS);
+  ACE_Arg_Shifter arg_shifter (argc, argv);
+
+  while (arg_shifter.is_anything_left())
+  {
+    // options:
+    //  -samples num_samples       defaults to infinite
+    //  -time num_seconds          defaults to infinite
+
+    const ACE_TCHAR *currentArg = 0;
+
+    if ((currentArg = arg_shifter.get_the_parameter(ACE_TEXT("-samples"))) != 0) {
```

Comment:
```suggestion
    if ((currentArg = arg_shifter.get_the_parameter(ACE_TEXT("--samples"))) != 0) {
```

Using double dashes gives use more flexibility later.